docs/source/node_operations.md

```diff
@@ -1,5 +1,108 @@
 # Node operations using Scylla Operator
 
+### Upgrading version of Scylla
+
+To upgrade Scylla version using Operator user have to modify existing ScyllaCluster definition.
+
+In this example cluster will be upgraded to `2020.1.0` version.
+```bash
+$ kubectl -n scylla patch ScyllaCluster simple-cluster  -p '{"spec":{"version": "4.2.2}}' --type=merge
+```
+
+Operator supports two types of version upgrades:
+1. Patch upgrade
+1. Generic upgrade
+
+
+**Patch upgrade**
+
+Patch upgrade is executed when only patch version change is detected according to [semantic versioning format](https://semver.org/).
+Procedure simply rolls out a restart of whole cluster and upgrades Scylla container image for each node one by one.
+
+Example: `4.0.0 -> 4.0.1`
+
+**Generic upgrade**
+
+Generic upgrades are executed for the non patch version changes.
+
+Example: `4.0.0 -> 2020.1.0` or `4.0.0 -> 4.1.0` or even `4.0.0 -> nightly`
+
+User can observe current state of upgrade in ScyllaCluster status.
+```bash
+$ kubectl -n scylla describe ScyllaCluster simple-cluster
+[...]
+Status:
+  Racks:
+    us-east-1a:
+      Members:        3
+      Ready Members:  3
+      Version:        4.1.9
+  Upgrade:
+    Current Node:         simple-cluster-us-east-1-us-east-1a-2
+    Current Rack:         us-east-1a
+    Data Snapshot Tag:    so_data_20201228135002UTC
+    From Version:         4.1.9
+    State:                validate_upgrade
+    System Snapshot Tag:  so_system_20201228135002UTC
+    To Version:           4.2.2
+```
+
+Each upgrade begins with taking a snapshot of `system` and `system_schema` keyspaces on all nodes in parallel.
+Name of this snapshot tag is saved in upgrade status under `System Snapshot Tag`.
+
+Before nodes in rack are upgraded, underlying StatefulSet is changed to use `OnDelete` UpgradeStrategy.
+This allows Operator have a full control over when Pod image is changed.
+
+When node is being upgraded, [maintenance mode](#maintenance-mode) is enabled, then node is being drained and snapshot of all data keyspaces is taken.
```

Comment:
"when node" -> "when a node", "then node is being drained" -> "then the node is drained"